src/main.cpp

```diff
@@ -5050,7 +5053,9 @@ bool ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv, int64_t
             if (!AlreadyHave(inv) && AcceptToMemoryPool(mempool, state, tx, true, &fMissingInputs))
             {
                 mempool.check(pcoinsTip);
-                RelayTransaction(tx);
+                std::vector<uint256> vAncestors;
+                mempool.queryAncestors(tx.GetHash(), vAncestors);
```

Comment:
Do we need to call queryAncestors if peer has not loaded a filter that wants ancestors? Or does it not matter performance wise?